_core/smart/overview.md

```diff
@@ -0,0 +1,135 @@
+
+## Basic building blocks of Clarity contracts
+
+The basic building blocks of Clarity are _atoms_ and _lists_. An atom is a number or string of contiguous characters. Some examples of atoms:
+
+* `token-sender`
+* `10000`
+* `SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR`
+
+Atoms can be native functions, user-defined functions, variables, and values that appear in a program. Functions that mutate data by convention terminate with an `!` exclamation point, for example the `insert-entry!` function. 
+
+A list is a sequences of atoms enclosed with `()` parentheses. Lists can contain other lists. Some examples of lists are:
+
+* `(get-block-info time 10)`
+* `(and 'true 'false)`
+* `(is-none? (get id (fetch-entry names-map (tuple (name \"blockstack\")))))`
+
+You can add comments to your Clarity contracts using  `;;` (double semi-colons). Both standalone and inline comments are supported.
+
+```cl
+;; Transfers tokens to a specified principal
+(define-public (transfer (recipient principal) (amount int))
+  (transfer! tx-sender recipient amount)) ;; returns: boolean
+```
+
+You use the `clarity-cli` command to check and launch a Clarity (`.clar`) program.
+
+## hello-world example
+
+The easiest program to run in any language is a hello world program. In Clarity, you can write this `hello-world.clar` program.
+
+```cl
+(begin 
+   (print "hello world"))
+```
+
+This program defines a single `hello-world` user function. The `begin` is a native Clarity function that evaluates the expressions input to it and returns the value of the last expression.  Here there is a single `print` expression. Both the `begin` and the `print` are enclosed in `()` parentheses.
+
+For the pre-release, the Blockstack virtual machine (VM) includes the `clarity-cli` command for interacting with the contract and SQLite to support the data space. You create a SQLLite database to hold data related to Clarity contracts. This database simulates the blockchain by recording the contract activity. 
+
+You can't run even an a hello-world program without first initializing a Clarity contract's data space within the database. You can use the `clarity-cli initialize` command to set up the database.
+
+```clarity-cli initialize /data/db```
+
+This command records the contract in the `db` database which resides in the `/data` directory of the VM. 
+You can name the database anything you like, the name `db` is not required. You can use SQLite to query this database:
+
+```sql
+sqlite> .open db
+sqlite> .tables
+contracts            maps_table           type_analysis_table
+data_table           simmed_block_table 
+sqlite> 
+```
+
+After you initialize the contract's data space, you can `check` a Clarity program for problems.
+
+```clarity-cli check ./hello.clar /data/db```
+
+As the name implies, the `check` ensures the contract definition passes a type check; passing programs return. Once a contract passes a check, you `launch` it.
```

Comment:
> passing programs return.

I think this should be "passing programs will return an exit code of 0"

Reply:
Thanks.